modules/apps/document-library/document-library-document-conversion/src/main/java/com/liferay/document/library/document/conversion/internal/DocumentConversionImpl.java

```diff
@@ -133,19 +140,82 @@ else if (!inputDocumentFormat.isExportableTo(outputDocumentFormat)) {
 		UnsyncByteArrayOutputStream unsyncByteArrayOutputStream =
 			new UnsyncByteArrayOutputStream();
 
-		DocumentConverter documentConverter = _getDocumentConverter();
+		AtomicBoolean isDone = new AtomicBoolean(false);
```

Comment:
You don't need the atomic boolean. Futures can return values; but in this case, it is not even necessary to do so.

To get rid of the atomic boolean you need to clearly separate the three possible outcomes of the future:
1. Success: this happens the call to `future.get()` returns normally.
2. Timeout: the timeout exn is thrown.
3. Any other random exception.

You're mixing 1 and 3; you should handle 3 similarly to two, with a catch block.

Reply:
Thanks!